addons/sale_project/views/project_task_views.xml

```diff
@@ -94,7 +94,7 @@
                 <field name="sale_order_id" string="Sale Order" filter_domain="['|', ('sale_order_id', 'ilike', self), ('sale_line_id', 'ilike', self)]"/>
             </xpath>
             <xpath expr="//search/group/filter[@name='customer']" position="after">
-                <filter string="Sales Order Item" name="sale_order_id" context="{'group_by': 'sale_order_id'}"/>
+                <filter string="Sales Order" name="sale_order_id" context="{'group_by': 'sale_order_id'}"/>
```

Comment:
The label is right, it is the field used that is wrong. 😇 
```suggestion
                <!-- Remove me in master -->
                <filter string="Sales Order" name="sale_order_id" context="{'group_by': 'sale_order_id'}" invisible="1"/>
                <filter string="Sales Order Item" name="sale_line_id" context="{'group_by': 'sale_line_id'}"/>
```
Thanks @tlibioulle

Reply:
LGTM :)